gpflow/optimizers/scipy.py

```diff
@@ -83,17 +86,19 @@ def _tf_eval(x: tf.Tensor) -> Tuple[tf.Tensor, tf.Tensor]:
         if jit:
             _tf_eval = tf.function(_tf_eval)
 
-        def _eval(x):
+        def _eval(x: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
looks like the result of `eval_func` is passed as the first arg to the scipy minimize function, which requires

>     fun : callable
        The objective function to be minimized.

            ``fun(x, *args) -> float``

        where x is an 1-D array with shape (n,) and `args`
        is a tuple of the fixed parameters needed to completely
        specify the function.

which doesn't appear to match with this signature. have i missed sth?

Reply:
I don't understand? there are no fixed parameters.

Reply:
the return type of this function is `(np.ndarray, np.ndarray)`, whereas the return type expected is `float`

Reply:
That's a flaw in the scipy documentation.